src/Models/Environment.php

```diff
@@ -0,0 +1,922 @@
+                [
+                    'method' => 'put',
+                    'form_params' => [
+                        'ssl_enabled' => false,
+                        'dedicated_ip' => false,
+                    ],
+                ]
+            );
+        } catch (\Exception $e) {
+            throw new TerminusException('There was an problem disabling https for this environment.');
+        }
+    }
+
+    /**
+     * Gives SFTP connection info for this environment
+     *
+     * @return array
+     */
+    public function sftpConnectionInfo()
+    {
+        if (!empty($ssh_host = $this->getConfig()->get('ssh_host'))) {
+            $username = "appserver.{$this->id}.{$this->site->id}";
+            $hostname = $ssh_host;
+        } elseif (strpos($this->getConfig()->get('host'), 'onebox') !== false) {
+            $username = "appserver.{$this->id}.{$this->site->id}";
+            $hostname = $this->getConfig()->get('host');
+        } else {
+            $username = "{$this->id}.{$this->site->id}";
+            $hostname = "appserver.{$this->id}.{$this->site->id}.drush.in";
+        }
+        $password = 'Use your account password';
+        $port = '2222';
+        $url = "sftp://$username@$hostname:$port";
+        $command = "sftp -o Port=$port $username@$hostname";
+        $info = [
+            'username' => $username,
+            'host' => $hostname,
+            'port' => $port,
+            'password' => $password,
+            'url' => $url,
+            'command' => $command,
+        ];
+        return $info;
+    }
+
+    /**
+     * "Wake" a site
+     *
+     * @return array
+     */
+    public function wake()
+    {
+        $on_stats = function (TransferStats $stats) {
+            $this->transfertime = $stats->getTransferTime();
+        };
+        $hostnames = $this->getHostnames()->ids();
+        $target = array_pop($hostnames);
+        $healthc = "http://$target/pantheon_healthcheck";
+        $response = $this->request()->request($healthc, compact('on_stats'));
+        $return_data = [
+            'success' => ($response['status_code'] === 200),
+            'time' => $this->transfertime,
+            'styx' => $response['headers']['X-Pantheon-Styx-Hostname'],
+            'response' => $response,
+            'target' => $target,
+        ];
+
+        return $return_data;
+    }
+
+    /**
+     * Deletes all content (files and database) from the Environment
+     *
+     * @return Workflow
+     */
+    public function wipe()
+    {
+        $workflow = $this->getWorkflows()->create('wipe');
+        return $workflow;
+    }
+
+    /**
+     * @return Backups
+     */
+    public function getBackups()
+    {
+        if (empty($this->backups)) {
+            $this->backups = $this->getContainer()->get(Backups::class, [['environment' => $this,]]);
+        }
+        return $this->backups;
+    }
+
+    /**
+     * @return \Terminus\Collections\Bindings
```

Comment:
Yeah. I've these keep getting eaten up by so I figured I'd go through and shorten them all up in one pass before merging the parent branch.